### DISCUSSION TITLE
Package publish check

### DISCUSSION THREAD

**Discussion by vincentvdk** [apple/pkl#644] [General]
Hi,

Whenever I want to build a new package I get the error that the package can not be fetched (remote location). This makes sense because this is a new version and the artifacts haven't been uploaded yet.

So what is the workflow, pass the `--skip-publish-check` for each new build? I'm a bit confused tbh.

Thanks

**Comment by bioball** [comment]
What's the full error message that you're getting? We don't intend on erroring if a package doesn't exist, and ignores HTTP 404 status codes.

**Comment by vincentvdk** [comment]
This is the output:

## Existing package
```
❯ pkl project package -e VERSION=0.0.2 config
Package `package://pkl-pkg.<redacted>/pkl-facilitator/pkl-facilitator@0.0.2` was already published with different contents.

Computed checksum: af61394f04e1a269553fc7054ac3acce09f6934e374d091eaff9d564efa9ebee
Published checksum: 70d325df1d1612075b391e512c22225c40f497ab8f92646c1acae552fa9e4e84

This likely means that you need to bump the packages version.
```

## New version
```
❯ pkl project package -e VERSION=0.0.3 config
Failed trying to fetch package `pkl-facilitator` from `https://pkl-pkg.<redacted>/pkl-facilitator/pkl-facilitator@0.0.3.zip` (response code `403`).
```

The packages are stored on S3 using AWS Cloudfront (caching disabled) for proper TLS termination.

**Comment by vincentvdk** [comment]
Well, because you mentioned the HTTP error code that is checked I noticed that Cloudfront is returning a `403` if the ` s3:ListBucket` permission is missing in the Bucket Policy.  It is now working after updating the policy.

Thanks for the pointer.

**Reply by bioball** [reply]
Good fix!
